src/java.base/share/classes/java/text/ChoiceFormat.java

```diff
@@ -249,86 +251,110 @@ public void applyPattern(String newPattern) {
      * further understanding of certain special characters: "#", "<", "\u2264", "|".
      */
     private void applyPatternImpl(String newPattern) {
-        StringBuilder[] segments = new StringBuilder[2];
-        for (int i = 0; i < segments.length; ++i) {
-            segments[i] = new StringBuilder();
-        }
-        double[] newChoiceLimits = new double[30];
-        String[] newChoiceFormats = new String[30];
-        int count = 0;
-        int part = 0; // 0 denotes limit, 1 denotes format
-        double startValue = 0;
-        double oldStartValue = Double.NaN;
+        Objects.requireNonNull(newPattern, "newPattern must not be null");
```

Comment:
I think this null check should be at the public method level, not the internal implementation. This technically ends up revealing the internal variable name (although it is the same with the public argument names as of now)